core/src/test/java/org/jdbi/v3/core/argument/TestBeanArguments.java

```diff
@@ -122,4 +122,95 @@ public void setBar(String bar) {}
         exception.expect(UnableToCreateStatementException.class);
         new BeanPropertyArguments("foo", bean).find("foo.bar", ctx);
     }
+
+    @Test
+    public void testBindNestedOptionalNull() throws Exception
```

Comment:
These are good tests to have, but we need to supplement them with tests that actually use the database, to confirm that optional properties work as expected.